tasks/office.py

```diff
@@ -66,34 +73,16 @@ def add_stored(self, object_key, object_data):
 
         if len(data_list) >= self.DATA_LIST_MAX:
             num_to_delete = random.randint(1, len(data_list))
-            del data_list[:num_to_delete]
+            # Convert a dict to a list so we can take a slice by insertion order fifo
+            self.local_store[object_key] = dict(list(data_list.items())[num_to_delete:])
 
         # Some creation endpoint auto-create multiple objects and return an array,
         # but each object in the array should still be considered individually here:
         if isinstance(object_data, list):
-            data_list.extend(object_data)
+            normalized = {value["id"]: value for value in object_data}
```

Comment:
Oooh I don't see this syntax often. It might be worth adding a comment about what's happening here.

Reply:
It's technically a dict comprehension but happy to add a comment too about the transformation.

Reply:
Maybe one comment for both of these lines that these are different examples of dict comprehension/composition. It's a neat example of two distinct strategies to do this!

Reply:
Done!